lib/Demangling/NodePrinter.cpp

```diff
@@ -169,13 +169,19 @@ static StringRef toString(ValueWitnessKind k) {
 
 class NodePrinter {
 private:
-  DemanglerPrinter Printer;
+  std::unique_ptr<DemanglerPrinter> PrinterStorage;
```

Comment:
Having two separate members feels a bit weird. Could we just make `Printer` a `std::unique_ptr<DemanglerPrinter>`? And take a `std::unique_ptr<DemanglerPrinter>` in the constructor? And if it's nullptr we just `make_unique`

Reply:
It didn't feel right yes. I initially wanted to pass an `std::optional`, but couldn't because it wouldn't allow me to pass in a child class. I will give it another go with your suggestion.